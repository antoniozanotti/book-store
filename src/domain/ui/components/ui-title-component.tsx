import { TitleSizeEnum } from "../enums/title-size-enum";
import { UiTitleComponentInterface } from "./ui-title-component-interface";
import React from "react";

export function UiTitleComponent({
  level = "h1",
  size = "x_large",
  children,
  ...props
}: UiTitleComponentInterface) {
  const baseStyles = "font-primary text-pretty";
  const sizeStyles = useSizeStyles(size);
  const headingClasses = `${baseStyles} ${sizeStyles}`;
  props.className = props.className
    ? `${headingClasses} ${props.className}`
    : headingClasses;

  return React.createElement(level, { ...props }, children);
}

function useSizeStyles(size: keyof typeof TitleSizeEnum) {
  switch (size) {
    case "small":
      return "text-2xl";
    case "medium":
      return "text-3xl";
    case "large":
      return "text-4xl sm:text-5xl md:text-6xl lg:text-7xl";
    default:
      return "text-5xl sm:text-6xl md:text-7xl lg:text-8xl";
  }
}
