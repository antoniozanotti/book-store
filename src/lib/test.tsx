import React, { ReactElement, ReactNode } from "react";
import {
  render,
  RenderOptions,
  RenderResult,
  Screen,
} from "@testing-library/react";

function AllTheProviders({ children }: { children: React.ReactNode }) {
  return <div data-testid="container">{children}</div>;
}

function customRender(
  ui: ReactElement,
  options?: Omit<RenderOptions, "wrapper">
): RenderResult {
  return render(ui, { wrapper: AllTheProviders, ...options });
}

function firstComponent(screen: Screen) {
  return screen.getByTestId("container").firstChild;
}

export * from "@testing-library/react";
export * from "@testing-library/user-event";
export { customRender as render, firstComponent };
import "@testing-library/jest-dom";
import { usePathname } from "next/navigation";

jest.mock("next/navigation", () => {
  return {
    useRouter() {
      return { push: () => {} };
    },
    useParams: () => ({}),
    notFound: () => null,
    usePathname: () => null,
    useSearchParams: () => null,
  };
});
