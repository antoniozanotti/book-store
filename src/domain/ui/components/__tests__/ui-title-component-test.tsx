import { render, screen } from "@/lib/test";
import { UiTitleComponent } from "../ui-title-component";

describe("UiTitleComponent", () => {
  it("should render p", async () => {
    const { container } = render(
      <>
        <UiTitleComponent level="p" size="small" className="line-clamp-2 mt-2">
          Test
        </UiTitleComponent>
        <UiTitleComponent level="h1" size="small">
          Stay Connected
        </UiTitleComponent>
        <UiTitleComponent level="h2" size="large">
          Featured Authors
        </UiTitleComponent>
        <UiTitleComponent size="large">Test</UiTitleComponent>
      </>
    );
    expect(container).toMatchSnapshot();
  });
});
