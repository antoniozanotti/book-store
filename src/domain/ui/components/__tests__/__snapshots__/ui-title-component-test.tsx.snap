// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UiTitleComponent should render p 1`] = `
<div>
  <div
    data-testid="container"
  >
    <p
      class="font-primary text-pretty text-2xl line-clamp-2 mt-2"
    >
      Test
    </p>
    <h1
      class="font-primary text-pretty text-2xl"
    >
      Stay Connected
    </h1>
    <h2
      class="font-primary text-pretty text-4xl sm:text-5xl md:text-6xl lg:text-7xl"
    >
      Featured Authors
    </h2>
    <h1
      class="font-primary text-pretty text-4xl sm:text-5xl md:text-6xl lg:text-7xl"
    >
      Test
    </h1>
  </div>
</div>
`;
